### PR TITLE
Add pre-commit hooks

### PR DESCRIPTION
**Pull request by chraibi**
With [pre-commit](https://pre-commit.com/) we can check if everything is well formated and mypy-checked before committing:
`
pre-commit run --all-files
`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -18,3 +18,4 @@ build==0.10
 mypy==1.4.1
 mypy-extensions==1.0
 ruff~=0.6
+pre-commit
```
Comment (schroedtert) [inline requirements.txt:21]:
```suggestion
pre-commit~=3.8
```

This way, we make sure to only use versions compatible with 3.8 (assuming you used the latest available version)

**Review by schroedtert — CHANGES_REQUESTED** [review]
I played with the idea of using pre-commit, nice that you did it!

But, then we also need to update the following:

- Dev Guide: How to set up and use it
- The CI script then should also use pre-commit. Otherwise, we have the same thing in two places, which is always bad

**Comment by chraibi** [comment]
> The CI script then should also use pre-commit. Otherwise, we have the same thing in two places, which is always bad

Yes, that is necessary. The pre-commit basically copies all checks of the ci. 
I don't want to mess up with Ci, though ^^

**Comment by schroedtert** [comment]
I can have a look and do it, if you want.

**Comment by chraibi** [comment]
> I can have a look and do it, if you want.

Sure! That would be nice!

**Comment by schroedtert** [comment]
@chraibi I updated some stuff, what do you think? Is it okay for you or is anything missing?

**Review by schroedtert — APPROVED** [review]

**Comment by chraibi** [comment]
> @chraibi I updated some stuff, what do you think? Is it okay for you or is anything missing?

Looks nice!

**Pull request merged**